apps/rush-lib/src/cli/utilities/pnpm/PnpmLinkManager.ts

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import * as semver from 'semver';
+import * as tar from 'tar';
+import encodeRegistry = require('encode-registry');
+
+import { JsonFile } from '@microsoft/node-core-library';
+
+import LinkManager, {
+  SymlinkKind
+} from '../LinkManager';
+import Utilities from '../../../utilities/Utilities';
+import Package from '../Package';
+import { RushConstants } from '../../../RushConstants';
+import { IRushLinkJson } from '../../../data/RushConfiguration';
+import RushConfigurationProject from '../../../data/RushConfigurationProject';
+
+export default class PnpmLinkManager extends LinkManager {
+  protected _linkProjects(): Promise<void> {
+    return new Promise<void>((resolve: () => void, reject: (reason: Error) => void): void => {
+      // go ahead and find out the registry we are using, we need this to locate the pnpm store folder
+      const registryUrl: string = Utilities.executeCommandAndCaptureOutput(
```

Comment:
>registryUrl [](start = 12, length = 11)

Validate this.

What if it returned multiple results?

Reply:
This should always return a single registry 

---
In reply to: [151792741](https://github.com/Microsoft/web-build-tools/pull/427#discussion_r151792741) [](ancestors = 151792741)